src/query/scan_manager.c

```diff
@@ -7730,9 +7730,10 @@ scan_print_stats_json (SCAN_ID * scan_id, json_t * scan_stats)
 
       if (scan_id->type == S_HEAP_SCAN)
 	{
-	  if (scan_id->scan_stats.agg_optimized_scan)
+	  if (scan_id->scan_stats.agg_index_name)
 	    {
-	      json_object_set_new (scan_stats, "aggregate optimized,", scan);
+	      json_object_set_new (scan, "alg_index", json_string (scan_id->scan_stats.agg_index_name));
+	      json_object_set_new (scan_stats, "agl", scan);
```

Comment:
Is it correct that "agl" is output in the json result?
```
{
  "Trace Statistics": {
    "SELECT": {
      "time": 0,
      "fetch": 0,
      "ioread": 0,
      "SCAN": {
        "access": "table (dba.t1)",
        "agl": { /********** this **********/
          "time": 0,
          "fetch": 0,
          "ioread": 0,
          "readrows": 0,
          "rows": 0,
          "alg_index": "pk_t1_id"
        }
      }
    }
  }
}
```
When printing as text, "agl" does not appear first. I think there should be consistency between the two output formats.
```
Trace Statistics:
  SELECT (time: 0, fetch: 0, ioread: 0)
    SCAN (table: dba.t1), (time: 0, fetch: 0, ioread: 0, readrows: 0, rows: 0, agl=pk_t1_id)
```

Reply:
Is there any way below?
It is not an index scan, but it is correct to use an index.
I think it's okay to display the bree time since we need to display the index name anyway.
Instead, all values will be zero.
```
Trace Statistics:
  SELECT (time: 0, fetch: 2, ioread: 0)
    SCAN (index: public.t1.pk_t1_id), (btree time: 0, fetch: 0, ioread: 0, readkeys: 0, filteredkeys: 0, rows: 0, covered: true, count_only: true)
```

I think it's okay to stick with "covered: true, count_only: true" as it is.
But if we really need to display "agl: true" then I don't think this is too bad.
```
Trace Statistics:
  SELECT (time: 0, fetch: 2, ioread: 0)
    SCAN (index: public.t1.pk_t1_id), (btree time: 0, fetch: 0, ioread: 0, readkeys: 0, filteredkeys: 0, rows: 0, agl: true)
```

Reply:
How about the below ?

Trace Statistics:
  SELECT (time: 0, fetch: 2, ioread: 0)
    SCAN (btree-header: dba.t1.pk_t1_id), (btree time: 0, fetch: 0, ioread: 0, readkeys: 0, filteredkeys: 0, rows: 0, agl: true)

Reply:
```
Trace Statistics:
  SELECT (time: 0, fetch: 0, ioread: 0)
    SCAN (table: public.t1), (heap time: 0, fetch: 0, ioread: 0, readrows: 0, rows: 0)
```

Information about the object used is separately written in `(table: public.t1)`.
The cumulative values and option information while scanning are written in `(heap time: 0, fetch: 0, ioread: 0, readrows: 0, rows: 0)`.

It is an opinion that `agl: dba.t1.pk_t1_id` should be written together with `(table: public.t1)` because it is the object information used.
And since it is index information, I think it should be written as `(index: public.t1.pk_t1_id)` instead of `(agl: public.t1.pk_t1_id)`.

`agl: true` is the option information to be used while scanning.
So I think it should be written as `(heap time: 0, fetch: 0, ioread: 0, readrows: 0, rows: 0)`.

The scan method is heap scan, but it uses index information.
So I think `btree time` is correct, not `heap time`.

I disagree with `btree-header` or `btree-agl` before `time`.
Since it is optional information used during scanning, I think it should be added at the end.
And if `btree time` is printed, `btree` does not have to be output redundantly in `btree-agl`.
We just need to print `agl`.

Reply:
I think `header_only: true` or `use_header: true` instead of `agl: true` is intuitive.

Reply:
fianally trace print, **"noscan"** shown
```
Trace Statistics:
    SELECT (time: 0, fetch: 2, ioread: 0)
        SCAN (noscan time: 0, fetch: 0, ioread: 0, readkeys: 0, filteredkeys: 0, rows: 0, agl: dba.t1.pk_t1_id)
```
Json format
```
"Trace Statistics": {
    "SELECT": {
      "time": 0,
      "fetch": 2,
      "ioread": 0,
      "SCAN": {
        "access": "table (dba.t1)",
        "noscan": {
          "time": 0,
          "fetch": 0,
          "ioread": 0,
          "readrows": 0,
          "rows": 0,
          "agl": "dba.pk_t1_id"
        }
      }
    }
  }  
```